src/index.ts

```diff
@@ -38,7 +38,11 @@ function yn (value: string | undefined) {
  * Debugging `ts-node`.
  */
 const shouldDebug = yn(process.env.TS_NODE_DEBUG)
-const debug = shouldDebug ? console.log.bind(console, 'ts-node') : () => undefined
+const timestamp = function(){};
+timestamp.toString = function(){
+  return "[ts-node " + (new Date).toISOString() + "]";
+}
```

Comment:
This looks unnecessarily clever.  Let's keep it simple and call the function every time we need a timestamp.  `function timestamp() {return "ts-node ...`